animation_nodes/nodes/geometry/intersect_line_line.py

```diff
@@ -1,36 +1,75 @@
     bl_idname = "an_IntersectLineLineNode"
     bl_label = "Intersect Line Line"
     bl_width_default = 160
-    searchTags = ["Closest Points on 2 Lines"]
+    searchTags = ["Nearest Points on 2 Lines"]
+
+    useFirstLineStartList = VectorizedSocket.newProperty()
+    useFirstLineEndList = VectorizedSocket.newProperty()
+    useSecondLineStartList = VectorizedSocket.newProperty()
+    useSecondLineEndList = VectorizedSocket.newProperty()
 
     def create(self):
-        self.newInput("Vector", "Line 1 Start", "lineStart1")
-        self.newInput("Vector", "Line 1 End", "lineEnd1", value = [0, 1, 0])
-        self.newInput("Vector", "Line 2 Start", "lineStart2")
-        self.newInput("Vector", "Line 2 End", "lineEnd2", value = [0, 0, 1])
-
-        self.newOutput("Vector", "Closest On Line 1", "closest1")
-        self.newOutput("Vector", "Closest On Line 2", "closest2")
-        self.newOutput("Boolean", "Is Valid", "isValid")
-
-    def getExecutionCode(self, required):
-        if len(required) == 0:
-            return
-
-        yield "closestPoints = mathutils.geometry.intersect_line_line(lineStart1, lineEnd1, lineStart2, lineEnd2)"
-
-        yield "if closestPoints is None:"
-        if "closest1" in required: yield "    closest1 = Vector((0, 0, 0))"
-        if "closest2" in required: yield "    closest2 = Vector((0, 0, 0))"
-        if "isValid" in required:  yield "    isValid = False"
-        yield "else:"
-        if "closest1" in required: yield "    closest1 = closestPoints[0]"
-        if "closest2" in required: yield "    closest2 = closestPoints[1]"
-        if "isValid" in required:  yield "    isValid = True"
-
-    def getUsedModules(self):
-        return ["mathutils"]
+        self.newInput(VectorizedSocket("Vector", "useFirstLineStartList",
+            ("First Line Start", "firstLineStart", dict(value = (1, 1, 0))),
+            ("First Lines Start", "firstLinesStart"),
+            codeProperties = dict(default = (1, 1, 0))))
+        self.newInput(VectorizedSocket("Vector", "useFirstLineEndList",
+            ("First Line End", "firstLineEnd", dict(value = (-1, -1, 0))),
+            ("First Lines End", "firstLinesEnd"),
+            codeProperties = dict(default = (-1, -1, 0))))
+
+        self.newInput(VectorizedSocket("Vector", "useSecondLineStartList",
+            ("Second Line Start", "secondLineStart", dict(value = (1, -1, 0))),
+            ("Second Lines Start", "secondLinesStart"),
+            codeProperties = dict(default = (1, -1, 0))))
+        self.newInput(VectorizedSocket("Vector", "useSecondLineEndList",
+            ("Second Line End", "secondLineEnd", dict(value = (-1, 1, 0))),
+            ("Second Lines End", "secondLinesEnd"),
+            codeProperties = dict(default = (-1, 1, 0))))
+
+        self.newOutput(VectorizedSocket("Vector", ["useFirstLineStartList", "useFirstLineEndList", "useSecondLineStartList", "useSecondLineEndList"],
```

Comment:
would be better to define something like `props = ["useFirstLineStartList", ...]` first and then just pass this to every socket to reduce duplication and line length.